db/design_doc.go

```diff
@@ -86,103 +86,91 @@ func (db *Database) PutDesignDoc(ddocName string, ddoc sgbucket.DesignDoc) (err
 
 const (
 	// viewWrapper_adminViews adds the rev to metadata, and strips the _sync property from the view result
-	viewWrapper_adminViews = `function(doc,meta) {
-	                    var sync = doc._sync;
-	                    if (sync === undefined || meta.id.substring(0,6) == "_sync:")
-	                      return;
-	                    if ((sync.flags & 1) || sync.deleted)
+	syncViewAdminWrapper = `function(doc,meta) {
```

Comment:
I think this could benefit quite a bit from some comments and slight formatting to aid understanding the next time somebody comes along to change this. Same for the user view wrapper below.

```go
    syncViewAdminWrapper = `
        function(doc,meta) {

            // skip doc IDs beginning with '_sync:'
            if(meta.id.substring(0,6) == "_sync:"){
                return;
            }

            var sync;
            var isXattr;

            // get sync data from either doc body or xattrs
            if (meta.xattrs === undefined || meta.xattrs.%s === undefined) {
                sync = doc._sync;
                isXattr = false;
            } else {
                sync = meta.xattrs.%s;
                isXattr = true;
            }

            // skip deleted documents
            if ((sync.flags & 1) || sync.deleted || sync === undefined) {
              return;
            }

            // strip _sync property from doc body before view
            if (!isXattr) {
                delete doc._sync;
            }

            // add rev to metadata
            meta.rev = sync.rev;
            (%s) (doc, meta);
            
            // re-insert sync data into doc body after view
            if (!isXattr) {
                doc._sync = sync;
            }
        }`
```